backend/app/api/routes/auth.py

```
@@ -1,226 +1,448 @@
+        raise
+    except Exception as e:
+        logger.error("Token refresh failed", error=str(e))
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="Invalid refresh token"
+        )
+
+
+@router.post("/logout", response_model=LogoutResponse)
+async def logout_user(
     credentials: HTTPAuthorizationCredentials = Depends(security),
```

Ruff (0.13.1)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)